File: src/main/java/rustleund/fightingfantasy/framework/closures/impl/RestoreScaleClosure.java
/*
 * Created on Oct 15, 2005
 */
package rustleund.fightingfantasy.framework.closures.impl;

import org.apache.commons.beanutils.PropertyUtils;
import org.w3c.dom.Element;

import rustleund.fightingfantasy.framework.base.GameState;
import rustleund.fightingfantasy.framework.base.Scale;

/**
 * @author rustlea
 */
public class RestoreScaleClosure extends AbstractClosure {

	private String scaleName = null;

	public RestoreScaleClosure(Element element) {
		this.scaleName = element.getAttribute("type");
	}

	@Override
	public boolean execute(GameState gameState) {

		Scale scale = null;

		try {
			scale = (Scale) PropertyUtils.getProperty(gameState.getPlayerState(), scaleName);
		} catch (Exception e) {
			e.printStackTrace();
		}

		scale.restorePreviousValue();

		return true;
	}

}

File: src/main/java/rustleund/fightingfantasy/framework/closures/impl/RestoreScaleClosure.kt
/*
 * Created on Oct 15, 2005
 */
package rustleund.fightingfantasy.framework.closures.impl

import org.apache.commons.beanutils.PropertyUtils
import org.w3c.dom.Element
import rustleund.fightingfantasy.framework.base.GameState
import rustleund.fightingfantasy.framework.base.Scale
import rustleund.fightingfantasy.framework.base.optionalAttribute
import rustleund.fightingfantasy.gamesave.SerializableClosure

/**
 * @author rustlea
 */
class RestoreScaleClosure(
    private val scaleName: String,
    private val label: String?
) : SerializableClosure {

    constructor(element: Element) : this(
        scaleName = element.getAttribute("type"),
        label = element.optionalAttribute("label")
    )

    override fun execute(gameState: GameState): Boolean {
        val scale = PropertyUtils.getProperty(gameState.playerState, scaleName) as Scale
        if (label == null) scale.restorePreviousValue() else scale.restoreSavedValue(label)
        return true
    }
}